File: core/src/main/java/heroes/journey/utils/art/TextureMap.java
package heroes.journey.utils.art;

public record TextureMap(String location, int width, int height) {

    public TextureMap(String location, int width, int height) {
        this.location = location.startsWith("Textures/") ? location : "Textures/" + location;
        this.width = width;
        this.height = height;
        ResourceManager.get().textureRegions.put(this, null);
    }

}

File: core/src/main/java/heroes/journey/utils/art/TextureMap.kt
package heroes.journey.utils.art

import heroes.journey.modlib.art.ITextureMap
import heroes.journey.modlib.registries.Registrable
import heroes.journey.mods.Registries

class TextureMap(id: String, location: String, override val width: Int, override val height: Int) :
    Registrable(id), ITextureMap {
    override val location: String = if (location.startsWith("Textures/")) location else "Textures/$location"

    override fun register(): TextureMap {
        ResourceManager.get().textureRegions[this] = null
        return Registries.TextureManager.register(this)
    }
}
